### PR TITLE
Typescriptifies websocket api

### PR DESCRIPTION
**Pull request by Pingviinituutti**

### CODE REVIEW

File: frontend/src/wsApi.ts [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
-  const onExternalDraw = useRef(() => {
-    console.error("No draw function set");
-  });
-  const onExternalWrite = useRef(() => {
-    console.error("No write function set");
-  });
-  const [readyState, setReadyState] = useState("CONNECTING");
+type EventTypes = "DRAWING_EVENTS" | "DRAWING_HISTORY" | "WRITE_HISTORY"
+
+interface DataInterface {
```
Comment (jhnj) [inline frontend/src/wsApi.ts:8]: No need to have `Interface` in the name IMO, `WebSocketData`?

File: frontend/src/wsApi.ts [hunk 2]
```diff
@@ -46,11 +59,11 @@ export const useWsApi = (url) => {
     });
 
     socket.current.addEventListener("message", (data) => {
-      const event = JSON.parse(data.data);
+      const event: DataInterface = JSON.parse(data.data);
```
Comment (jhnj) [inline frontend/src/wsApi.ts:62]: There is no validation here right? Just typecasting?

Comment (Pingviinituutti) [reply frontend/src/wsApi.ts:62]: Correct

File: frontend/src/wsApi.ts (hunk 1 above)
Comment (Pingviinituutti) [reply frontend/src/wsApi.ts:8]: As long as it does not create any mixups with other functions, variables etc.

**Comment by thesauri** [comment]
- Fixed Prettify errors
- Renamed `DataInterface` to `WebSocketPayload`, this should be a bit clearer what it refers to

**Pull request merged**